libexec/rtld-elf/riscv/reloc.c

```diff
@@ -129,8 +129,9 @@ _rtld_relocate_nonplt_self(Elf_Dyn *dynp, Elf_Auxinfo *aux)
 	caprelocslim = (const struct capreloc *)((const char *)caprelocs + caprelocssz);
 	pcc = __builtin_cheri_program_counter_get();
 	/* TODO: allow using tight bounds for RTLD */
-	_do___caprelocs(caprelocs, caprelocslim, relocbase, pcc,
-	    (Elf_Addr)relocbase, false);
+	cheri_init_globals_impl(caprelocs, caprelocslim,
```

Comment:
I guess this is fine for as long as rtld isn't using sub-library compartments for itself (😱)

Reply:
I think we believe that rtld will not have compartments.  In particular though, we don't at this point have the bits in place to permit separate code capabilities.  We'd have to find the phdr and enumerate the compartments, etc.  If we ever do that, then we will have to duplicate the code to deal with caprelocs.

That is all true btw if we want to support compartments in static binaries in that in the csu code we'd have to implement it by hand using the proper PCC cap there as well.